Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp

```diff
@@ -547,6 +547,12 @@ namespace AZ
                     return;
                 }
 
+                for (auto& dependency : jobProduct.m_dependencies)
```

Comment:
Please leave a comment to explain the reason this is being added, that there's a potential deadlock and briefly describe how it happens.